product_pack/views/product_pack_line_views.xml

```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2019 Tecnativa - Ernesto Tejeda
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+    <record model="ir.ui.view" id="product_pack_line_form">
+        <field name="name">product.pack.line.form</field>
+        <field name="model">product.pack.line</field>
+        <field name="type">form</field>
+        <field name="arch" type="xml">
+            <form string="Pack Product Line">
+                <group>
+                    <field name="product_id"/>
+                    <field name="quantity"/>
+                    <field name="discount" groups="sale.group_discount_per_so_line" invisible="context.get('pack_type') == 'fixed_price'"/>
```

Comment:
But I see a bit confusing to use both specific price unit and discount. If you want an specific price, put in on pack line price unit, and if you want to apply a discount over the corresponding sales price  (after applying customer pricelist), then that's the way to go, but not both...

IMO, we should remove specific price, put the discount on the sales part, and follow this last strategy. What do you think?

Reply:
The problem with specific prices is that they are not maintainable over time (you can't modify them easily on batch). And it's a very rough edge, as the usual case is to define a price in the pack product, not to specify discounts per product. That's why I think it's better even to leave out both for now.

Reply:
Thank you very much for the video! Yeah, now I have it more clear, and of course I have mixed the `price_unit` at sale line pack level with product pack line.

Any way, you can see that you have packed a lot of different features for a lot of use cases, and I would unpack them, unfolding these features through successive modules. For the "assisted" feature (which is not a correct English name IMO for what you have showcased me in the video, but I don't have right now a better one), it's clearly an extension module for sale_product_pack, adding the option in pack type and incorporating the logic in sales order, so better to rip off now. What are the advantages of doing so? We will get smaller pieces that can be migrated, tested, etc faster. We avoid also possible side effects, cleaner code (we avoid a lot of `if` in the same base code). Do you agree in this unfolding? (we will take care).

About the discount, let's keep it, but on `sale_product_pack` (and properly labelled that is for sales), because if we share the same discount for sales and purchases, on merchandise (purchase and sell) businesses, we won't want to have same discount on both operations.